Include Documentation for Ethdo Wallet Integration (#248)
prysmaticlabs/documentation: website/docs/wallet/ethdo.md

---
id: ethdo
title: Integrating with third-party wallet, ethdo
sidebar_label: Using Ethdo
---

This section explains how to use the popular, third-party wallet, [ethdo](https://github.com/wealdtech/ethdo/tree/v1.6.1) with Prysm's validator client. At the time of writing, this guide is made for ethdo v1.6.1. Ethdo offers a command-line tool that allows advanced stakers to perform all sorts of tasks relating to wallets and validating keys which is far more extensible than what Prysm has built-in. Ethdo provides useful helpers for managing HD wallets, signing deposits, performing validator exits, and more. In this guide, we'll go over how to use a wallet you created in ethdo to validate with Prysm.

:::danger This is the only recommended way to use Ethdo with Prysm

> This guide is the only recommended way to use ethdo with Prysm which will always be supported by our client. Do not use ethdo with the old flags --keymanager, --keymanageropts, or --disable-accounts-v2. These flags will be removed in Prysm's beta release

Reviewer: ```suggestion
This guide is the only recommended way to use ethdo with Prysm which will always be supported by our client. Do not use ethdo with the old flags --keymanager, --keymanageropts, or --disable-accounts-v2. These flags will be removed in Prysm's beta release!
```

:::

## Installation

> This is recommended option to keep using ethdo for all your wallet needs and simply import the validating keys you wish to validate with in Prysm.

Reviewer: ```suggestion
This is the recommended option to keep using ethdo for all your wallet needs and simply import the validating keys you wish to validate with in Prysm.
```

This sentence is hard to read in general. Please consider rephrasing.

To begin, follow the instructions to fetch and install Prysm for your operating system.

* [Using the Prysm installation script (Recommended)](/docs/install/install-with-script)
* [Using Docker](/docs/install/install-with-docker)
* [Building from source with Bazel (Advanced)](/docs/install/install-with-bazel)

Next, make sure you install [ethdo](https://github.com/wealdtech/ethdo/tree/v1.6.1).

### Create an ethdo wallet

> If you already have an ethdo wallet, you can skip this step. You can follow along the [usage documentation](https://github.com/wealdtech/ethdo/blob/master/docs/usage.md#create-1) in ethdo itself for the most up-to-date information on available options, but at the most basic level, you can create an ethdo wallet as follows:

Reviewer: ```suggestion
If you already have an ethdo wallet, you can skip this step. You can follow along the [usage documentation](https://github.com/wealdtech/ethdo/blob/v1.6.1/docs/usage.md#create-1) in ethdo itself for the most up-to-date information on available options, but at the most basic level, you can create an ethdo wallet as follows:
```
Pin to `v1.6.1`
